### PR TITLE
[vimpothr] Add new extractor

### PR DESCRIPTION
**Pull request by dotWee**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This pull request adds support for extracting videos hosted on [vimp.oth-regensburg.de/video/...](https://vimp.oth-regensburg.de). Vimp is the main hosting platform for lectures and videos from the [OTH Regensburg](https://oth-regensburg.de) in Germany.  

Currently only basic fields are supported with more to be implemented soon. An extractor for videos hosted on `vimp.oth-regensburg.de/m/...` will also be added soon.

Personal TODO:

- [ ] Possible to-extract-able informations
    - [ ] `uploader`
    - [ ] `license`
    - [ ] `creator`
    - [ ] `channel`
    - [ ] `channel_id`
    - [ ] `channel_url`
    - [ ] `view_count`
    - [ ] `like_count`
    - [ ] `categories`

### CODE REVIEW

File: youtube_dl/extractor/vimpothr.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            'description': 'Invarianzsatz, Rentenbarwerte',
+            'thumbnail': 'https://vimp.oth-regensburg.de/cache/b56c79d598c594b117d769dc1731ed6a.jpg',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(r'<h1>(.+?)</h1>', webpage, 'title') or self._og_search_title(webpage)
```
Comment (dirkf) [inline youtube_dl/extractor/vimpothr.py:25]: `_html_search_regex()` throws unless given a `default`(returns it silently) or `fatal=False` (returns `None` with a warning), so the OG search won't execute.

Comment (dotWee) [reply youtube_dl/extractor/vimpothr.py:25]: I'm wondering if it would make sense to rely only on _og_search. What would you recommend?

Comment (dirkf) [reply youtube_dl/extractor/vimpothr.py:25]: It depends what content you typically find in the first <h1> element vs the content attribute of the <meta thingy=og:title> element.

But maybe you could actually ask someone at OTH what their policy on metadata is. If you got a useful answer, you could treat what they say as the first target and add fall-backs for any of your test videos that don't comply.

**Comment by FestplattenSchnitzel** [comment]
@dotWee with https://github.com/yt-dlp/yt-dlp/pull/3604 merged, the master of [yt-dlp](https://github.com/yt-dlp/yt-dlp) should now be able to support vimp.oth-regensburg.de. (You can install it with `pip install git+https://github.com/yt-dlp/yt-dlp.git`.)

**Comment by dotWee** [comment]
@FestplattenSchnitzel thanks for pinging me! Just tested it with a random public video, works as expected!

**Comment by dirkf** [comment]
It looks like we should back-port the yt-dlp extractor. Is there anything in this PR that should be added to it?

**Pull request closed without merging**